staging/src/k8s.io/client-go/discovery/cached/disk/cached_discovery.go

```diff
@@ -63,14 +63,16 @@ var _ discovery.CachedDiscoveryInterface = &CachedDiscoveryClient{}
 // ServerResourcesForGroupVersion returns the supported resources for a group and version.
 func (d *CachedDiscoveryClient) ServerResourcesForGroupVersion(groupVersion string) (*metav1.APIResourceList, error) {
 	filename := filepath.Join(d.cacheDirectory, groupVersion, "serverresources.json")
-	cachedBytes, err := d.getCachedFile(filename)
+	cachedBytes, cacheErr := d.getCachedFile(filename)
 	// don't fail on errors, we either don't have a file or won't be able to run the cached check. Either way we can fallback.
-	if err == nil {
+	if cacheErr == nil {
 		cachedResources := &metav1.APIResourceList{}
 		if err := runtime.DecodeInto(scheme.Codecs.UniversalDecoder(), cachedBytes, cachedResources); err == nil {
 			klog.V(10).Infof("returning cached discovery info from %v", filename)
 			return cachedResources, nil
 		}
```

Comment:
Can you add a V(6).Infof to report the decoding error as well?

Reply:
Thanks!  Done :)

Reply:
Thanks. Done :)